### PR TITLE
Created 6.x samples for server client and shared contract.

### LINKED ISSUE #25071: .NET 6.x update for Code-first gRPC services topic
Update for .NET 6.0.

Still refers to Startup.cs for example.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 548ea859-02aa-913e-a2bb-e75d45d45f75
* Version Independent ID: 50d09935-a98d-86ae-7196-4d9a55f9121e
* Content: [Code-first gRPC services and clients with .NET](https://docs.microsoft.com/en-us/aspnet/core/grpc/code-first?view=aspnetcore-6.0)
* Content Source: [aspnetcore/grpc/code-first.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/grpc/code-first.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-grpc**
* GitHub Login: @JamesNK
* Microsoft Alias: **jamesnk**

### PR DESCRIPTION
**Pull request by wadepickett**
[Internal Review](https://review.docs.microsoft.com/en-us/aspnet/core/grpc/code-first?view=aspnetcore-6.0&branch=pr-en-us-25124)

Fixes #25071

.NET 6.x update for:
- Topic:  Code-first gRPC services and clients with .NET
- Server, Client and Shared Contract app samples

### CODE REVIEW

**Comment by wadepickett** [comment]
@JamesNK, I updated for 6.0 for the code sample and added .csproj and more of the program.cs context.   Is there anything else you would like changed or added for 6.0?

File: aspnetcore/grpc/code-first.md [hunk 1]
```diff
@@ -74,20 +83,26 @@ gRPC services implemented with code-first and `.proto` files can co-exist in the
...
+* Add a reference to the shared code-contract project.
 
-The preceding code:
+[!code-csharp[](code-first/samples/6.x/GrpcGreeterClient/GrpcGreeterClient.csproj?highlight=10-13,15-17)]
+
+The preceding gRPC client `.csproj` file highlights the required package and shared-code project references.
+
+[!code-csharp[](code-first/samples/6.x/GrpcGreeterClient/Program.cs?highlight=13,15-16)]
+
+The preceding gRPC client `program.cs` code:
```
Comment (serpent5) [inline aspnetcore/grpc/code-first.md:97]:
```suggestion
The preceding gRPC client `Program.cs` code:
```

File: aspnetcore/grpc/code-first.md [hunk 2]
```diff
@@ -54,18 +61,20 @@ To add gRPC code-first service to an ASP.NET Core app:
 * Add a [protobuf-net.Grpc.AspNetCore](https://www.nuget.org/packages/protobuf-net.Grpc.AspNetCore) package reference.
 * Add a reference to the shared code-contract project.
 
-Create a new `GreeterService.cs` file and implement the `IGreeterService` service interface:
+  [!code-csharp[](code-first/samples/6.x/GrpcGreeter/GrpcGreeter.csproj?highlight=9-11,13-15)]
```
Comment (serpent5) [inline aspnetcore/grpc/code-first.md:64]:
```suggestion
  [!code-xml[](code-first/samples/6.x/GrpcGreeter/GrpcGreeter.csproj?highlight=9-11,13-15)]
```

File: aspnetcore/grpc/code-first.md [hunk 3]
```diff
@@ -74,20 +83,26 @@ gRPC services implemented with code-first and `.proto` files can co-exist in the
...
 * Add a [protobuf-net.Grpc](https://www.nuget.org/packages/protobuf-net.Grpc) package reference.
-* Add a reference to the shared code-contract project.
+
 * Add a [Grpc.Net.Client](https://www.nuget.org/packages/Grpc.Net.Client) package reference.
 
-[!code-csharp[](code-first/samples/5.x/GrpcGreeterClient/Program.cs?name=snippet&highlight=2,4-5)]
+* Add a reference to the shared code-contract project.
 
-The preceding code:
+[!code-csharp[](code-first/samples/6.x/GrpcGreeterClient/GrpcGreeterClient.csproj?highlight=10-13,15-17)]
```
Comment (serpent5) [inline aspnetcore/grpc/code-first.md:91]:
```suggestion
[!code-xml[](code-first/samples/6.x/GrpcGreeterClient/GrpcGreeterClient.csproj?highlight=10-13,15-17)]
```

**Review by serpent5 — APPROVED** [review]
Looks good. Just a couple of minor suggestions inline.

You could also consider switching to file-scoped namespaces in the code samples. I've been doing that in 6.0 samples.

**Comment by wadepickett** [comment]
@JamesNK, I will go ahead and merge this update.  If there was any additional info you wanted for 6.0 or other changes, let me know and I will track with a new issue and update immediately.

**Pull request merged**